code/clue.js

```diff
@@ -1,59 +1,357 @@
+  weight: 4,
+  image: 'assets/candlestick.png',
+  id: 'candlestick'
+};
+
+const dumbbell = {
+  name: 'Dumbbell',
+  weight: 12,
+  image: 'assets/dumbbell.png',
+  id: 'dumbbell'
+};
+
+const poison = {
+  name: 'Poison',
+  weight: 0.5,
+  image: 'assets/poison.png',
+  id: 'poison'
+};
+
+const axe = {
+  name: 'Axe',
+  weight: 15,
+  image: 'assets/axe.png',
+  id: 'axe'
+};
+
+const bat = {
+  name: 'Bat',
+  weight: 9,
+  image: 'assets/bat.png',
+  id: 'bat'
+};
+
+const trophy = {  
+  name: 'Trophy',
+  weight: 4,
+  image: 'assets/trophy.png',
+  id: 'trophy'
+};
+
+const pistol = {
+  name: 'Pistol',
+  weight: 8,
+  image: 'assets/pistol.png',
+  id: 'pistol'
+};
+
+// Room objects
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const diningRoom = {
+  name: 'Dining Room',
+  image: 'assets/dining-room.png'
+}; 
+  
+const conservatory = {
+  name: 'Conservatory',
+  image: 'assets/conservatory.png' 
+};
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const kitchen = {
+  name: 'Kitchen',
+  image: 'assets/kitchen.png'
+};
+
+const study = {
+  name: 'Study',
+  image: 'assets/study.png' 
+};
+
+const library = {
+  name: 'Library',
+  image: 'assets/library.png'
+};
+
+const billiardRoom = {
+  name: 'Billiard Room',
+  image: 'assets/billiard-room.png'
+};
+
+const lounge = {
+  name: 'Lounge',
+  image: 'assets/lounge.png'
+};
+
+const ballroom = {
+  name: 'Ballroom',
+  image: 'assets/ballroom.png'
+};
+
+const hall = {
+  name: 'Hall', 
+  image: 'assets/hall.png'
+};
+
+const spa = {
+  name: 'Spa',
+  image: 'assets/spa.png'
+};
+ 
+const livingRoom = {
+  name: 'Living Room',
+  image: 'assets/living-room.png'
+};
+
+const observatory = {
+  name: 'Observatory',
+  image: 'assets/observatory.png'
+};
+
+const theater = {
+  name: 'Theater',
+  image: 'assets/theater.png'
+};
+ 
+ const guestHouse = {
+   name: 'Guest House',
+   image: 'assets/guest-house.png'
+ };
+
+ const patio = {
+   name: 'Patio',
+   image: 'assets/patio.png'
+ };
+
+// Arrays
 
 const suspects = [
   mrGreen,
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
   mrsWhite
-  // ...  and the rest
-]
+];
 
-const weapons = []
+const weapons = [
+  rope, 
+  knife, 
+  candlestick, 
+  dumbbell, 
+  poison, 
+  axe, 
+  bat, 
+  trophy, 
+  pistol
+];
 
-const rooms = []
+const rooms = [
+  diningRoom, 
+  conservatory, 
+  kitchen, 
+  study, 
+  library, 
+  billiardRoom, 
+  lounge, 
+  ballroom, 
+  hall, 
+  spa, 
+  livingRoom, 
+  observatory, 
+  theater,
+  guestHouse,
+  patio
+];
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+  return array[Math.floor(Math.random() * array.length)];
+};
+
+// Mystery object
+let mystery = {
+  killer: '',
+  weapon: '',
```

Comment:
It's even more semantic approach to initialise mystery object with `null` values like this :

```
const mystery = {
  killer: null,
  room: null,
  weapon: null
}
```

Moreover, you can use `const` keyword - we will be changing values of properties, that's true, but JavaScript will not consider this as overwriting. What counts for JavaScript is that object will stay object